libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/BuildCacheRelocationIT.kt

```diff
@@ -36,7 +36,6 @@ class BuildCacheRelocationIT : BaseGradleIT() {
     override fun defaultBuildOptions(): BuildOptions =
         super.defaultBuildOptions().copy(
             withBuildCache = true,
-            androidGradlePluginVersion = AGPVersion.v3_6_0,
```

Comment:
Currently we rely on` BuildOptions.androidGradlePluginVersion` to determine whether it is an android test and what AGP version it is using.

Set Agp version only for those android tests to allow other non-android test run with warning-mode=fail if possible.

Reply:
Do we need this change, as it seems that all tests in this class are running with `--warning-mode=fail`?